src/include/sof/math/trig.h

```diff
@@ -89,18 +123,35 @@ static inline int32_t sin_fixed_32b(int32_t th_rad_fxp)
  * | 32 | 28  |  1    | 32 | 31 |   1	| 4.28	 | 1.31   |
  * +------------------+-----------------+--------+--------+
  */
+#ifdef CORDIC_HIFI3
+static inline int32_t cos_fixed_32b(int32_t th_rad_fxp)
+{
+	ae_int32 th_cdc_fxp;
+	ae_int32 b_yn;
+	ae_int32 xn;
+	ae_int32 sign;
+
+	cordic_approx(th_rad_fxp, CORDIC_31B_TABLE_SIZE, &sign, &b_yn, &xn, &th_cdc_fxp);
+	th_cdc_fxp = sign * xn;
+
+	/*convert Q2.30 to Q1.31 format*/
+	return (ae_int32)AE_SLAI32S(th_cdc_fxp, 1);
+}
+#else
```

Comment:
In this specific situation, is the return statement the only real change? Feels like some unwarranted code duplication to me.

Reply:
Yep, the ae_int types are needed to avoid type casts.